### PR TITLE
[APIM] Add Named Value operations

### PR DESCRIPTION
**Pull request by syneex**
**Description**  

This PR contains code that adds Named Value operations to the az apim command module. The following operations were added:

- apim nv create
- apim nv update
- apim nv delete
- apim nv show
- apim nv list

**Testing Guide**  
I have extended the testing scenario contained in the apim command module folder. For testing execute `azdev test apim` in the virtual environment. The testing sequence I implemented is described here:
1. First I create a named value
2. Then I get the named value
3. After these two operations i update one Named Value
4. The fourth step contains listing of all named values (which should equal a count of one)
5. And at the end I delete that Named Value.

**History Notes**  
[APIM] Added general az apim named value commands

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
APIM

File: src/azure-cli/azure/cli/command_modules/apim/_params.py [hunk 1]
```diff
@@ -114,6 +114,31 @@ def load_arguments(self, _):
         c.argument('subscription_required', arg_group='API', arg_type=get_three_state_flag(), help='If true, the API requires a subscription key on requests.')
         c.argument('tags', tags_type)
 
+    with self.argument_context('apim nv create') as c:
+        c.argument('service_name', options_list=['--service-name'], help='The name of the API Management service instance.')
+        c.argument('named_value_id', help='The id of the API Management Named Value.')
```
Comment (MyronFanQiu) [inline src/azure-cli/azure/cli/command_modules/apim/_params.py:119]: Which kinds of id? GUID?

File: src/azure-cli/azure/cli/command_modules/apim/_params.py [hunk 2]
```diff
@@ -114,6 +114,31 @@ def load_arguments(self, _):
...
+        c.argument('display_name', help='The display name of the API Management Named Value.')
+        c.argument('value', help='The value of the API Management Named Value.')
+        c.argument('secret', arg_type=get_three_state_flag(), help='If true, the Named Value is a secret.')
+        c.argument('tags', tags_type)
+
+    with self.argument_context('apim nv update') as c:
```
Comment (MyronFanQiu) [inline src/azure-cli/azure/cli/command_modules/apim/_params.py:125]: You can put `c.argument('service_name')` under `with self.argument_context('apim nv')`, and we can share it across all sub command.

Comment (MyronFanQiu) [reply src/azure-cli/azure/cli/command_modules/apim/_params.py:125]: maybe just delete the left part and rename `apim nv create` to `apim nv`.

**Comment by MyronFanQiu** [comment]
@syneex  Based on the error message, you might need new python sdk. Please bump the version of the dependency. Better to run all existing tests lively to avoid breaking change in python sdk.

**Comment by RupengLiu** [comment]
@syneex thanks for your contribution, I have taken over your pr along with adding product entity commands

File: src/azure-cli/azure/cli/command_modules/apim/_params.py (hunk 1 above)
Comment (syneex) [reply src/azure-cli/azure/cli/command_modules/apim/_params.py:119]: Not necessarily. Could be a normal name, a GUID or some random string.

File: src/azure-cli/azure/cli/command_modules/apim/_params.py (hunk 2 above)
Comment (syneex) [reply src/azure-cli/azure/cli/command_modules/apim/_params.py:125]: okay I will do so.

**Comment by MyronFanQiu** [comment]
@syneex Nice to see you back. @RupengLiu Could you discussed with @syneex. It seems like that you did same work in another PR.

**Comment by syneex** [comment]
I could open a new PR and pull the master from azure cli before that so I have the current status. Then there would be no need for discussion with @RupengLiu

**Comment by MyronFanQiu** [comment]
@syneex It seems that your work relies on a new version of python sdk. It's related with bumping dependency which will affect other apimanagement commands as well. I think @RupengLiu can help you on this since he is the owner of `az apim` right now.

**Comment by syneex** [comment]
@MyronFanQiu Can I close this PR? I saw the PR from @RupengLiu regarding the Named Values and its way better than mine.

**Comment by MyronFanQiu** [comment]
@syneex Sure. Thanks for your contribution!

**Pull request closed without merging**